### ISSUE TITLE
GZIP doesn't work with got

### ISSUE THREAD

**Issue by mikicho** [mswjs/interceptors#446]
```js
const { ClientRequestInterceptor } = require('@mswjs/interceptors/ClientRequest')
const got = require('got')
const zlib = require('zlib')

const interceptor = new ClientRequestInterceptor({
  name: 'my-interceptor',
})
interceptor.apply();
interceptor.on('request', ({ request }) => {
  const compressed = zlib.gzipSync('hello')
  request.respondWith(new Response(compressed, {
    status: 200, headers: {
      'X-Transfer-Length': String(compressed.length),
      'Content-Length': undefined,
      'Content-Encoding': 'gzip',
    }
  }))
});

(async function () {
  console.log(1);
  console.log(await got('http://example.test/foo'))
  console.log(2);
})()
```
Specifically, got exits [in this line](https://github.com/sindresorhus/got/blob/v11.8.6/source/core/utils/get-buffer.ts#L9) with error code 0 (ok).
I couldn't catch the error (even with a try-catch), so I'm unsure what's happening.

It does work with native `http`.

May be related to: #308

**Comment by mikicho** [comment]
I investigated it further.
It seems like we don't emit a `readable` event. The [`for await (const chunk of stream)` ](https://github.com/sindresorhus/got/blob/v11.8.6/source/core/utils/get-buffer.ts#L9) in `got` waits for a `readable` event, which is never emitted.
I suspect it is connected to the [response clones](https://github.com/mswjs/interceptors/issues/443#issuecomment-1739123449).

**Comment by kettanaito** [comment]
## Released: v0.36.6 🎉

This has been released in v0.36.6!

- 📄 [**Release notes**](https://github.com/mswjs/interceptors/releases/tag/v0.36.6)
- 📦 [npm package](https://www.npmjs.com/package/@mswjs/interceptors/v/0.36.6)

Make sure to always update to the latest version (`npm i @mswjs/interceptors@latest`) to get the newest features and bug fixes.

---

_Predictable release automation by [@ossjs/release](https://github.com/ossjs/release)_.
